support/src/test.rs

```diff
@@ -0,0 +1,23 @@
+// KILT Blockchain – https://botlabs.org
+// Copyright (C) 2019-2023 BOTLabs GmbH
+
+// The KILT Blockchain is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The KILT Blockchain is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+// If you feel like getting in touch with us, you can do so at info@botlabs.org
+
+use scale_info::prelude::{boxed::Box, string::String};
+
+pub fn convert_error_message(error_message: String) -> &'static str {
+	Box::leak(error_message.into_boxed_str())
```

Comment:
I agree the naming of the function is not ideal, as is not using `test` as the module name. Nevertheless, this is feature-gated, hence it's not possible to leak data in the runtime itself. Or did I not fully understand your point?

Reply:
Even leaking memory in tests is not something i would like to introduce so lightly. How many testcases do we have? how many iterations inside a test case do we do? Do we really want to end up trying to debug why our tests use so much memory?

My point is that we add a pitfall to our code base. If you are not careful and call `convert_error_message` too often we might get memory issues. The test program doesn't exit once an error is reported. And this method might get called even if no error is reported and the test was successful.

Reply:
Between an error message that just tells you something went wrong and requires you to gather more information yourself, and one that gives you already enough information to know where to look for the problem, but that could leak some memory, I would lean for the second option, which is also what I suggested Adel. `Box::leak` was one solution I found to have `&static str` that actually contain useful information. If there's a better alternative, or if you think this should not be in the codebase at all, or something else, is eventually up to you, I guess.
I just liked to have a `Delegator 4q4aTGAhfa43R1G4i3DaYMMDsDkgu2CfqAiwUup87KjFTV6M is expected to have 3 KILTs staked, but 2 KILTs were found` over `Delegator stake does not match the expected one`.

Reply:
I log the detailed error message with `log::error!` and return for all sanity test a generic error string `"Sanity test error"`. Otherwise, I have to do something like: 

`pub fn log_and_return_error_message(detailed_error_message: String, error_message: &'static str) -> &'static str {
	log::error!("{}", detailed_error_message);
        error_message
}
`

Which does not feel right.